File: Backend/src/Sofco.Model/Models/Admin/Functionality.cs
using System.Collections.Generic;
using Sofco.Model.Interfaces;
using Sofco.Model.Relationships;

namespace Sofco.Model.Models.Admin
{
    public class Functionality : BaseEntity, ILogicalDelete
    {
        public string Description { get; set; }
        public bool Active { get; set; }
        public string Code { get; set; }

        public int ModuleId { get; set; }
        public Module Module { get; set; }

        public IList<RoleFunctionality> RoleFunctionality { get; set; }
    }
}

File: Frontend/src/models/admin/functionality.ts
import { Module } from "models/admin/module";

export class Functionality {
    constructor(
        public id: number,
        public code: string,
        public description: string,
        public active: boolean,
        public startDate: Date,
        public endDate: Date,
        public modules: Module[]
    ){}

}